File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans=0,mn=INT_MAX;
        for(int x:prices)
        {
            mn=min(x,mn);
            ans=max(ans,x-mn);
        }
        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        ans=0
        mn=prices[0]
        for x in prices:
           mn=min(mn,x)
           ans=max(ans,x-mn)
        return ans;
        